pasta/base/annotate.py

```diff
@@ -1027,13 +1019,52 @@ class AstAnnotator(BaseVisitor):
   def __init__(self, source):
     super(AstAnnotator, self).__init__()
     self.tokens = token_generator.TokenGenerator(source)
+    self._indent = ''
+    self._indent_diff = ''
 
   def visit(self, node):
     try:
+      ast_utils.setprop(node, 'indent', self._indent)
+      ast_utils.setprop(node, 'indent_diff', self._indent_diff)
       super(AstAnnotator, self).visit(node)
     except (TypeError, ValueError, IndexError, KeyError) as e:
       raise AnnotationError(e)
 
+  def indented(self, node, children_attr):
+    """Annotate children with their indentation level and iterate over them."""
+    children = getattr(node, children_attr)
+    cur_loc = self.tokens._loc
+    next_loc = self.tokens.peek().start
+    # Special case: if the first child is on the same line, then there is no
```

Comment:
first and only? this doc doesnt seem to correspodn to `len(children) == 1 `to me or did you mean >=?

Reply:
Actually, it should just be first. It also applies to the following:

```python
if a: b; c
```

which has two child nodes of the `If` node.

```
>>> ast.dump(ast.parse('if a: b; c'))
"Module(body=[If(test=Name(id='a', ctx=Load()), body=[Expr(value=Name(id='b', ctx=Load())), Expr(value=Name(id='c', ctx=Load()))], orelse=[])])"
```